main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -447,8 +420,8 @@ object SetFormula {
     case Compl(f1) => f1
     case inter@Inter(_, _, _, _, _, _, _) =>
       mkUnionAll(inter.mapSubformulas(mkCompl))
-    case union@Union(_, _, _, _, _, _, _) =>
-      mkInterAll(union.mapSubformulas(mkCompl))
+    case Union(_) =>
+      Compl(f)
```

Comment:
This also partially changes how complement works, before this PR negation is pushed to the bottom. I think it'd be the most proper to a) remove normal-negation-form from mkCompl b) disable propagation and then c) do this PR. This would keep each PR sensible and see the effect of each change individually

Reply:
Perhaps, but then performance would dip. I removed union while keeping and adding enough stuff to keep performance the same.